### PR TITLE
Passing TestPlatform properties to DataCollector extensions

### PR DESCRIPTION
**Pull request by vagisha-nidhi**

### CODE REVIEW

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+}
+```
+
+At present, we will be adding `testSources` to `sessionStartEventArgs.Properties` which can be taken use by the datacollector.
+
+In the datacollector, the user can get test sources as given below.
+```csharp
+IEnumerable<string> sources = new List<string>();
+sources = sessionStartEventArgs.GetPropertyValue("testSources", sources);
```
Comment (acesiddhu) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: IEnumerable<string> sources = new List<string>();
sources = sessionStartEventArgs.GetPropertyValue<IEnumerable<string>>("testSources", sources);

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    ValidateArg.NotNull(property, "property");
+
+    if (value != null)
+    {
+        this.Properties.Add(property, value);
+    }
+}
+```
+
+At present, we will be adding `testSources` to `sessionStartEventArgs.Properties` which can be taken use by the datacollector.
```
Comment (acesiddhu) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: which can be used by data collector

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# 0026 DataCollector Extensibility - Getting testSources in property bag
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: Something like - Passing TesPlatform properties to datacollector extensions

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# 0026 DataCollector Extensibility - Getting testSources in property bag 
+
+## Summary
+Getting `testSources` as part of property bag. List of test sources can be used by datacollectors for processing before test run start.
+
+## Motivation
+Data collector might need test sources list while initialization. Example: Static code coverage data collector needs to instrument the test sources before test run start.
+
+In `sessionStartEventArgs` the field `IDictionary<string, object> Properties` will contain all properties we need to pass to the datacollector.
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: capital S

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# 0026 DataCollector Extensibility - Getting testSources in property bag 
+
+## Summary
+Getting `testSources` as part of property bag. List of test sources can be used by datacollectors for processing before test run start.
+
+## Motivation
+Data collector might need test sources list while initialization. Example: Static code coverage data collector needs to instrument the test sources before test run start.
+
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: Add design heading

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md (hunk 4 above)
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: SessionStartEventArgs will contains a test platform properties bag. Currently TestPlatform passes following properties to the datacollector extensions:
1. "TestSources" : IEnumerable<String>
    TestSources is enumberable of testSources passed by used in test run.

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+/// </summary>
+/// <param name="property">
+/// Name of the property
+/// </param>
+/// <param name="value">
+/// Value of the property
+/// </param>
+public void SetPropertyValue(string property, object value)
+{
+    ValidateArg.NotNull(property, "property");
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: remove the implementation from RFC

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    if (value != null)
+    {
+        this.Properties.Add(property, value);
+    }
+}
+```
+
+At present, we will be adding `testSources` to `sessionStartEventArgs.Properties` which can be taken use by the datacollector.
+
+In the datacollector, the user can get test sources as given below.
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: Add heading for `Usage`

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+/// <summary>
+/// Returns the property value
+/// </summary>
+/// <param name="value">
+/// Value of the property
+/// </param>
+/// <param name="property">
+/// Name of the property
+/// </param>
+public T GetPropertyValue<T>(string property, T value)
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: remove value argument

File: RFCs/0026-DataCollector Extensibility - Getting testSources.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+/// Returns all TestPlatform Properties currently specified passed to the datacollector extenstion
+/// </summary>
+public IEnumerator<KeyValuePair<string, object>> GetProperties()
+
+/// <summary>
+/// Returns the property value corresponding to the given property name.
+/// </summary>
+/// <param name="property"> Name of the property </param>
+public T GetPropertyValue<T>(string property)
+
```
Comment (abhishkk) [inline RFCs/0026-DataCollector Extensibility - Getting testSources.md]: add GetPropertyValue with object also.

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
+# 0026 DataCollector Extensibility - Passing TestPlatform properties to DataCollector extensions 
+
+## Summary
+Passing TestPlatform properties as part of property bag to datacollectors. These properties can be used by the datacollectors for processing before test run start.
+
+## Motivation
+Data collector might need test platform properties while initialization. Example: Static code coverage data collector needs to instrument the test sources before test run start thus need list of test sources during initialization.
```
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: dotn use intialization

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+## Design
+`SessionStartEventArgs` will contain `Properties` which are passed to the data collector in `SessionStart` event.
+Currently test platform passes following properties to the datacollector extensions :
+1. **"TestSources"** : IEnumerable
```
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: IEnumberable<String>

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    TestSources is an enumerable of all test sources that is used by the test run.
```
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: In view, its not coming in separate line

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md (hunk 11 above)
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: we have only one property. So remove 1.

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+The public APIs exposed in `SessionStartEventArgs` would be as given below.
+```csharp
+ /// <summary>
+/// Returns all TestPlatform Properties currently specified passed to the datacollector extenstion
```
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: comments

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+/// </summary>
+/// <param name="property"> Name of the property </param>
+public object GetPropertyValue(string property)
+
+```
+
+## Usage
+In the datacollector, the user can get test sources as given below.
+```csharp
+sources = args.GetPropertyValue<IEnumerable>("TestSources");
```
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: IEnumberale<string> sources = args.GetPropertyValue<IEnumberale<string>>("TestSources");

File: RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+Also, all properties can be accessed via the `GetProperties` API as given.
+```csharp
+var Properties = args.GetProperties();
```
Comment (abhishkk) [inline RFCs/0026-Passing-TestPlatform-Properties-to-DataCollector.md]: nit: small p

**Pull request merged**